### PR TITLE
WF17 initial impl

### PR DESCRIPTION
**Pull request by robstryker**
Signed-off-by: Rob Stryker <rob@oxbeef.net>

### CODE REVIEW

**Comment by robstryker** [comment]
We don't actually have a wf17 instance to test against yet but it should work fine

**Comment by robstryker** [comment]
testPR

**Comment by robstryker** [comment]
testPR

File: api/docs/org.jboss.tools.rsp.schema/src/main/resources/client/typescript/src/protocol/generated/protocol.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@
  */
 export namespace Protocol {
     /* tslint:disable */
-    // Generated using typescript-generator version 2.2.413 on 2019-06-03 14:27:41.
+    // Generated using typescript-generator version 2.2.413 on 2019-06-05 13:54:35.
```
Comment (jeffmaury) [inline api/docs/org.jboss.tools.rsp.schema/src/main/resources/client/typescript/src/protocol/generated/protocol.ts]: This change should be removed from the PR for better clarity

File: api/docs/org.jboss.tools.rsp.schema/src/main/resources/schema/typescript/protocol.unified.d.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /* tslint:disable */
-// Generated using typescript-generator version 2.2.413 on 2019-06-03 14:27:41.
+// Generated using typescript-generator version 2.2.413 on 2019-06-05 13:54:35.
```
Comment (jeffmaury) [inline api/docs/org.jboss.tools.rsp.schema/src/main/resources/schema/typescript/protocol.unified.d.ts]: This change should be removed from the PR for better clarity

File: runtimes/bundles/org.jboss.tools.rsp.server.wildfly/src/main/java/org/jboss/tools/rsp/server/wildfly/servertype/capabilities/WildFlyExtendedProperties.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+ ******************************************************************************/ 
+package org.jboss.tools.rsp.server.wildfly.servertype.capabilities;
+
+import org.jboss.tools.rsp.server.spi.servertype.IServer;
+import org.jboss.tools.rsp.server.wildfly.servertype.launch.Wildfly100DefaultLaunchArguments;
+import org.jboss.tools.rsp.server.wildfly.servertype.launch.Wildfly110DefaultLaunchArguments;
+import org.jboss.tools.rsp.server.wildfly.servertype.launch.Wildfly80DefaultLaunchArguments;
+
+public class WildFlyExtendedProperties {
+	public static class Wildfly80ExtendedProperties extends AbstractWildflyExtendedProperties {
```
Comment (adietish) [inline runtimes/bundles/org.jboss.tools.rsp.server.wildfly/src/main/java/org/jboss/tools/rsp/server/wildfly/servertype/capabilities/WildFlyExtendedProperties.java]: sonar: add a private constructor to prohibit instantiation

File: runtimes/bundles/org.jboss.tools.rsp.server.wildfly/src/main/java/org/jboss/tools/rsp/server/wildfly/servertype/capabilities/WildFlyExtendedProperties.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+		public Wildfly80ExtendedProperties(IServer server) {
+			super("8.x", "1.7", "1.8", "service:jmx:http-remoting-jmx", new Wildfly80DefaultLaunchArguments(server), server);
+		}
+	}
+	public static class Wildfly90ExtendedProperties extends AbstractWildflyExtendedProperties {
+		public Wildfly90ExtendedProperties(IServer server) {
+			super("9.x", "1.7", "1.8", "service:jmx:remote+http", new Wildfly80DefaultLaunchArguments(server), server);
```
Comment (adietish) [inline runtimes/bundles/org.jboss.tools.rsp.server.wildfly/src/main/java/org/jboss/tools/rsp/server/wildfly/servertype/capabilities/WildFlyExtendedProperties.java]: sonar: extract ```"service:jmx:remote+http"``` to a constant

File: api/docs/org.jboss.tools.rsp.schema/src/main/resources/client/typescript/src/protocol/generated/protocol.ts (hunk 1 above)
Comment (adietish) [reply api/docs/org.jboss.tools.rsp.schema/src/main/resources/client/typescript/src/protocol/generated/protocol.ts]: +1, but then I think that this new protocol.ts should not even be included in this PR that's about providing support for a new server.

File: api/docs/org.jboss.tools.rsp.schema/src/main/resources/schema/typescript/protocol.unified.d.ts (hunk 2 above)
Comment (adietish) [reply api/docs/org.jboss.tools.rsp.schema/src/main/resources/schema/typescript/protocol.unified.d.ts]: see above (same here, should not be included)

**Comment by adietish** [comment]
> We don't actually have a wf17 instance to test against yet but it should work fine

there is a Beta1 here: https://download.jboss.org/wildfly/17.0.0.Beta1/wildfly-17.0.0.Beta1.zip

File: runtimes/bundles/org.jboss.tools.rsp.server.wildfly/src/main/java/org/jboss/tools/rsp/server/wildfly/servertype/capabilities/WildFlyExtendedProperties.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+		}
+	}
+	public static class Wildfly120ExtendedProperties extends AbstractWildflyExtendedProperties {
+		public Wildfly120ExtendedProperties(IServer server) {
+			super("12.0", "1.8", "10.", "service:jmx:remote+http", new Wildfly100DefaultLaunchArguments(server), server);
+		}
+	}
+	public static class Wildfly130ExtendedProperties extends AbstractWildflyExtendedProperties {
+		public Wildfly130ExtendedProperties(IServer server) {
+			super("13.0", "1.8", "11.0", "service:jmx:remote+http", new Wildfly110DefaultLaunchArguments(server), server);
```
Comment (adietish) [inline runtimes/bundles/org.jboss.tools.rsp.server.wildfly/src/main/java/org/jboss/tools/rsp/server/wildfly/servertype/capabilities/WildFlyExtendedProperties.java]: I think that "11.0" should be "11."

File: runtimes/bundles/org.jboss.tools.rsp.server.wildfly/src/main/java/org/jboss/tools/rsp/server/wildfly/servertype/capabilities/WildFlyExtendedProperties.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+		}
+	}
+	public static class Wildfly140ExtendedProperties extends AbstractWildflyExtendedProperties {
+		public Wildfly140ExtendedProperties(IServer server) {
+			super("14", "1.8", "11.0", "service:jmx:remote+http", new Wildfly110DefaultLaunchArguments(server), server);
```
Comment (adietish) [inline runtimes/bundles/org.jboss.tools.rsp.server.wildfly/src/main/java/org/jboss/tools/rsp/server/wildfly/servertype/capabilities/WildFlyExtendedProperties.java]: I think that "11.0" should be "11."

**Comment by robstryker** [comment]
testPR

**Comment by adietish** [comment]
I can confirm that Wildfly 17 works with the code that was added in this PR. Tested it with the final release of wildfly 17 which I downloaded here: https://download.jboss.org/wildfly/17.0.0.Final/wildfly-17.0.0.Final.zip

**Pull request merged**